pkg/service/admission/endpoint_admission.go

```diff
@@ -0,0 +1,115 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+
+	kadmission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+const RestrictedEndpointsPluginName = "RestrictedEndpointsAdmission"
+
+func init() {
+	kadmission.RegisterPlugin("RestrictedEndpointsAdmission", func(client clientset.Interface, config io.Reader) (kadmission.Interface, error) {
```

Comment:
Why do we need to register if we are explicitly adding and initializing the plugin
(https://github.com/danwinship/origin/blob/13cc03e2f32c23521035a4ac1ba0bc157e834e07/pkg/cmd/server/kubernetes/master_config.go#L60 and https://github.com/danwinship/origin/blob/13cc03e2f32c23521035a4ac1ba0bc157e834e07/pkg/cmd/server/kubernetes/master_config.go#L192)